CollapseLauncher/Classes/Helper/Image/Waifu2X.cs

```diff
@@ -0,0 +1,181 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace CollapseLauncher.Helper.Image
+{
+    public class Waifu2X : IDisposable
+    {
+        public const string DllName = "Lib\\waifu2x-ncnn-vulkan";
+
+        [DllImport(DllName)]
+        private static extern IntPtr waifu2x_create(int gpuId = 0, bool ttaMode = false, int numThreads = 1);
+
+        [DllImport(DllName)]
+        private static extern void waifu2x_destroy(IntPtr context);
+
+        [DllImport(DllName)]
+        private static extern unsafe int waifu2x_load(IntPtr context, byte* param, byte* model);
+
+        [DllImport(DllName)]
+        private static extern unsafe int waifu2x_process(IntPtr context, int w, int h, int c, byte* inData, byte* outData);
+
+        [DllImport(DllName)]
+        private static extern unsafe int waifu2x_process_cpu(IntPtr context, int w, int h, int c, byte* inData, byte* outData);
+
+        [DllImport(DllName)]
+        private static extern void waifu2x_set(IntPtr context, Param param, int value);
+
+        [DllImport(DllName)]
+        private static extern int waifu2x_get(IntPtr context, Param param);
+
+        [DllImport(DllName)]
+        [return:MarshalAs(UnmanagedType.I1)]
+        private static extern bool waifu2x_support_gpu(IntPtr context);
+
+        public enum Param
+        {
+            Noise,
+            Scale,
+            TileSize,
+        }
+
+        public enum Waifu2XStatus
+        {
+            Ok,
+            CpuMode,
+            NotAvailable,
+            TestNotPassed,
+        }
+
+        private IntPtr _context;
+        private byte[] _paramBuffer;
+        private byte[] _modelBuffer;
+        private bool _testPassed;
+
+        public Waifu2XStatus Status
+        {
+            get
+            {
+                // Usability testing
+                if (_context == 0)
+                    return Waifu2XStatus.NotAvailable;
+                else if (!_testPassed)
+                    return Waifu2XStatus.TestNotPassed;
+
+                if (waifu2x_support_gpu(_context))
+                    return Waifu2XStatus.Ok;
+                else
+                    return Waifu2XStatus.CpuMode;
+            }
+        }
+
+        public Waifu2X(int gpuId = 0, bool ttaMode = false, int numThreads = 1)
+        {
+            try
+            {
+                _context = waifu2x_create(gpuId, ttaMode, numThreads);
```

Comment:
As per discussion internally, please add debug/info log around when w2x is invoked as it can take a bit of time, so the user that seeing the console can understand that its actually doing it instead of the image failed to load.